Add docs for cubeviz spatial region selection (#351)
spacetelescope/jdaviz: docs/cubeviz/displaycubes.rst

desired region is covered by a colored shape, after which you release the mouse button.
Once you have completed the action of clicking and dragging, there is an opportunity to reposition
the region of interest by clicking on it and moving the region to a different location in the image viewer.
If you are satisfied with the positioning of your region of interest, simply click another part of the
viewer to lock it in place. Once a region is selected, the cube will be collapsed in wavelength space over the region,
and the resulting spectrum will be displayed in the 1d spectrum viewer at the bottom of the UI.

.. image:: img/subset_creation.png

There are other options available for region of interest. At the top of the User Interface,
there is a section that says either "+ No selection (create new)" or "Subset n" where n is an integer
that tells you which Subset is currently selected. To the right of this area, are red circles that allow
you to change the method of region selection. The options are `replace`, `add`, `and`, `xor`, and `remove`.
You can use these options to further adjust the region of interest and adapt it to your use case.

> 

Reviewer: I would like a description of how to use each option--describe what replace, add, and and xor do when you have them selected and either create a new region or modify an existing one.

to the left or right in order to get a lower or higher slice number, respectively. Another option is to
click the :guilabel:`circle` that controls the location on the :guilabel:`slider` and use the
:guilabel:`left` or :guilabel:`right` arrow keys on your keyboard to find a slice number with more precision.
A third option is to type a slice number into the :guilabel:`text box` to the right of the :guilabel:`slider`
and press :guilabel:`enter` on your keyboard. A fourth option is to use the :guilabel:`up/down arrows` inside
the :guilabel:`text box` to raise/lower the slice number.

Defining Spatial Regions
========================

Regions can be created by first clicking the :guilabel:`hammer and screwdriver` icon and
selecting either the :guilabel:`rectangular region of interest` icon or the
:guilabel:`circular region of interest` icon, then clicking and dragging (using the mouse) until the
desired region is covered by a colored shape, after which you release the mouse button.
Once you have completed the action of clicking and dragging, there is an opportunity to reposition

> the region of interest by clicking on it and moving the region to a different location in the image viewer.

Reviewer 2: ```suggestion
the region of interest by dragging the region to a different location in the image viewer.
```

I read this as a full KeyOn and KeyOff event, followed by dragging the mouse with no buttons pressed. I thought it was a click-and-drag event in order to move it? If I'm mistaken, feel free to ignore this one!

Author: I'm not exactly sure what the question is. You click and drag to create the region, which is completed once the mouse button is released. The user then has an opportunity to reposition the region by clicking on it once again and moving (dragging) the region to a different location. Once the user clicks off of the region, it is locked.

Author: I think I understand now, I'll switch the wording to "clicking and dragging the region"

If you are satisfied with the positioning of your region of interest, simply click another part of the

> viewer to lock it in place. Once a region is selected, the cube will be collapsed in wavelength space over the region,

Reviewer 2: > simply click another part of the viewer to lock it in place.

Oh really? You have to actually click somewhere else in order for the moved region to stay? I feel like the user should just be able to let go and it should just work? I'd suggest we should change this behavior later

Author: Yes, I think the idea is that you should be able to move the region around before locking it in place, so it does not lock until you click elsewhere in the image viewer.

Reviewer: The existing behavior allows you to continue editing a region with the selected logical function until you click elsewhere. 
Maybe don't require the click and the region remains editable until another subset is generated?  However, that would
mean autogenerating the collapse after each sub-region edit.
